src/hooks/use-event-analytic.tsx

```diff
@@ -0,0 +1,26 @@
+import {useCallback} from 'react';
+import {verifyHooksExist} from './verify-hooks-exist';
+import {AnalyticsDispatcher} from 'shisell';
+import {useAnalytics} from './use-analytics';
+
+export function useEventAnalytic<EventType extends React.SyntheticEvent>(
+    eventHandler: React.EventHandler<EventType> | null | undefined,
```

Comment:
maybe change the order? `(dispatch, eventHandler?, deps?)`
because `dispatch` is not optional, you always have to pass it, and `eventHandler` and `deps` are optional

Reply:
Good point